src/com/palmergames/bukkit/towny/object/Nation.java

```diff
@@ -204,7 +206,32 @@ else if (town.hasNation())
 		}
 	}
 
-	public void setCapital(Town capital) {
+	public void setCapital(Town capital) throws TownyException {
+		if (isCapital(capital))
+			return;
+		
+		NationCapitalChangeEvent nationChange = new NationCapitalChangeEvent(this, capital, true);
+		NationKingChangeEvent kingChangeEvent = new NationKingChangeEvent(this, capital.getMayor(), true);
+		Bukkit.getPluginManager().callEvent(nationChange);
+		Bukkit.getPluginManager().callEvent(kingChangeEvent);
+		if (nationChange.isCancelled() || kingChangeEvent.isCancelled())
+			throw new TownyException(TownySettings.getLangString("msg_err_new_capital_cancelled"));
+
+		this.capital = capital;
+		try {
+			recheckTownDistance();
+			TownyPerms.assignPermissions(capital.getMayor(), null);
+		} catch (Exception e) {
+			// Dummy catch to prevent errors on startup when setting nation.
+		}
+	}
+
+	public void forceSetCapital(Town capital) {
+		if (isCapital(capital))
+			return;
+		
+		Bukkit.getPluginManager().callEvent(new NationCapitalChangeEvent(this, capital, false));
+		Bukkit.getPluginManager().callEvent(new NationKingChangeEvent(this, capital.getMayor(), false));
```

Comment:
Since these methods have pretty much the same code, I would overload them so that one method accepts a boolean cancellable parameter, while the other is a default method that calls the overloaded method with cancellable passed in as true.
E.g.
```java
setCapital(Town capital) {
    setCapital(capital, true);
}

setCapital(Town capital, boolean cancellable) {
...
}
```

Reply:
I've implemented it, but with the default being a non-forced call. Let me know your thoughts on that.